service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadDocumentsService.java

```diff
@@ -0,0 +1,117 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.CourtBundle;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentMetaData;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentSocialWorkOther;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.findElement;
+
+@Service
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class UploadDocumentsService {
+
+    private final Time time;
+    private final DocumentUploadHelper documentUploadHelper;
+
+    public Map<String, Object> updateCaseDocuments(CaseData caseDataBefore,
+                                                   CaseData caseData) {
+        List<Element<DocumentSocialWorkOther>> listOfOtherDocs = setUpdatedByAndDateAndTimeOnDocuments(
+                caseData.getOtherSocialWorkDocuments(), caseDataBefore.getOtherSocialWorkDocuments());
+
+        Document socialWorkChronologyDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkChronologyDocument(), caseDataBefore.getSocialWorkChronologyDocument());
+        Document socialWorkStatementDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkStatementDocument(), caseDataBefore.getSocialWorkStatementDocument());
+        Document socialWorkAssessmentDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkAssessmentDocument(), caseDataBefore.getSocialWorkAssessmentDocument());
+        Document socialWorkCarePlanDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkCarePlanDocument(), caseDataBefore.getSocialWorkCarePlanDocument());
+        Document socialWorkEvidenceTemplateDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getSocialWorkEvidenceTemplateDocument(),
+                caseDataBefore.getSocialWorkEvidenceTemplateDocument());
+        Document thresholdDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getThresholdDocument(), caseDataBefore.getThresholdDocument());
+        Document checklistDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getChecklistDocument(), caseDataBefore.getChecklistDocument());
+        CourtBundle courtBundleDocument = setUpdatedByAndDateAndTimeOnDocuments(
+            caseData.getCourtBundle(), caseDataBefore.getCourtBundle());
```

Comment:
@lewisbirks do you think there's a nicer way for doing this?

Reply:
Not sure if there is a nicer way but it could definitely be formatted nicer

Reply:
Something like with a gap between each call and the parenthesis spread out. Parameters can be on the same or different line, all that really matters is that there is clear separation between each method call.
```java
        Document thresholdDocument = setUpdatedByAndDateAndTimeOnDocuments(
            caseData.getThresholdDocument(),
            caseDataBefore.getThresholdDocument()
        );
​
        Document checklistDocument = setUpdatedByAndDateAndTimeOnDocuments(
            caseData.getChecklistDocument(),
            caseDataBefore.getChecklistDocument()
        );
```

Reply:
done